### DISCUSSION TITLE
-statistic Gradient not giving the results I expect (I think lol)

### DISCUSSION THREAD

**Discussion by jarj1** [ImageMagick/ImageMagick#2492] [Help]
I have a simple picture two rectangles one above is yellow with a Saturation of 63.6% and a peach one below with a Saturation of 25%

![yellow](https://user-images.githubusercontent.com/66709392/91790990-60e1af80-ec55-11ea-9ea7-f19b520bbd58.png)

https://www.imagemagick.org/script/command-line-options.php#statistic 
Tells me the "Gradient   =  maximum difference (max - min) value in neighborhood"

So from the command:
 "magick yellow.png -alpha off  -colorspace HSV -channel Saturation -statistic Gradient   3x1  -grayscale RMS yellowSG3x1.png" 
I expect a "black/0" image because when comparing 3 pixels horizontally they would all have the same Saturation and so the difference between the min and max is 0
![yellowSG3x1](https://user-images.githubusercontent.com/66709392/91790993-62ab7300-ec55-11ea-87ce-239bc0e4ec20.png)

Likewise:
 "magick yellow.png -alpha off  -colorspace HSV -channel Saturation -statistic Gradient   1x3  -grayscale RMS yellowSG1x3.png" 
I expect a "black/0" image with a two pixel high band in the middle representing the junction between the two rectangles and the 38.6% difference between the min and max Saturation.
![yellowSG1x3](https://user-images.githubusercontent.com/66709392/91790991-6212dc80-ec55-11ea-9ab8-61cef0ac43de.png)
The yellowSG1x3.png image gives 2 X 1 pixel bands of different differences when IMHO they should both be the same and either side of the band is not black/0 as I expected it to be

I'm probably misunderstanding something or left something out of the command - help guidance much appreciated

**Comment by fmw42** [comment]
There is no -channel saturation. Try channel 1 or channel g. Separate channels and remove your -grayscale, since the channel will be gray when separated.

`magick yellow.png -alpha off -colorspace HSV -channel 1 -separate +channel -statistic Gradient 3x1 yellowSG3x1.png`

![yellowSG3x1](https://user-images.githubusercontent.com/13458218/91797860-7664c400-ebd8-11ea-887d-62fc6ae67f9e.png)

See https://imagemagick.org/script/command-line-options.php#channel
